### PR TITLE
Feature/cc 0114 show quotation list

### PR DESCRIPTION
**Pull request by Double-Twelve**
When admin log in successfully, he/she can go to the quotation page, all the quotations will be displayed, and can search quotation list for a specific account

image

### CODE REVIEW

**Comment by ShawnWang86b** [comment]
error scroll
1 error scroll appear
2 error box design could discuss

**Comment by ShawnWang86b** [comment]
image

**Review by STY1997 — COMMENTED** [review]
TQL...

**Comment by ShawnWang86b** [comment]
image

File: src/redux/api/quotationApi.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export const quotationApi = createApi({
+  reducerPath: "quotation",
+  baseQuery: fetchBaseQuery({
+    baseUrl: environment.apiBaseUrl,
+  }),
+  endpoints: (builder) => ({
+    getAllQuotation: builder.query<
+      { data: IQuotation[]; total: number },
+      { offset: number; limit: number; optionalQuery: string }
```
Comment (ShawnWang86b) [inline src/redux/api/quotationApi.tsx:13]: optionalQuery   in api 
`&user_id=${searchValue}&sort=${sorting[0]?.id}&desc=${sorting[0]?.desc ? -1: 1}`  
and handle search which uses lodash
is this search one required user ID's data directly from the backend or for other purposes?
not quite understand, can you explain when you have time, thx

File: src/components/DisplayDataTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+            <Thead backgroundColor="#2C4E8A">
+              {table.getHeaderGroups().map((headerGroup) => (
+                <Tr key={headerGroup.id}>
+                  {headerGroup.headers.map((header) => {
+                    return (
+                      <Th
+                        key={header.id}
+                        fontSize="12px"
+                        color="#FFFFFF"
+                        cursor={header.column.getCanSort() ? "pointer" : "default"}
```
Comment (ShawnWang86b) [inline src/components/DisplayDataTable/index.tsx:203]: when clicking the header's word, the position of the word will change, and in my opinion, it should have a sort related to time, update tim
image
e or create time. and not sure the reason quotation and quotation detail no sort function.

Comment (ShawnWang86b) [reply src/components/DisplayDataTable/index.tsx:203]: let's discuss when you have time

**Review by ShawnWang86b — APPROVED** [review]

**Comment by Double-Twelve** [comment]
> error scroll 1 error scroll appear 2 error box design could discuss

Fixed issue with scrollbars appearing

File: src/components/DisplayDataTable/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Dispatch, SetStateAction } from "react";
+import { InputGroup, InputLeftElement, Input, InputRightElement } from "@chakra-ui/react";
+import { HiOutlineSearch, HiOutlineX } from "react-icons/hi";
+
+interface Props {
+  searchPlaceholder: string;
+  searchValue: string;
+  setSearchValue: Dispatch<SetStateAction<string>>;
```
[the remark this answers is not in the document]
Comment (Double-Twelve) [reply src/components/DisplayDataTable/Search.tsx:8]: change to _useRef_

**Review by wwwseven — APPROVED** [review]

**Pull request merged**